lib/ansible/modules/web_infrastructure/ansible_tower/tower_workflow_launch.py

```diff
@@ -0,0 +1,163 @@
+    type: string
+job_info:
+    description: dictionnary containing information about the workflow executed
+    returned: If workflow launched
+    type: dict
+'''
+
+
+EXAMPLES = '''
+- name: Launch a workflow
+  tower_workflow_launch:
+    name: "Test Workflow"
+  delegate_to: localhost
+  run_once: true
+  register: workflow_results
+
+- name: Launch a Workflow with parameters without waiting
+  tower_workflow_launch:
+    workflow_template: "Test workflow"
+    extra_vars: "---\nmy: var"
+    wait: False
+  delegate_to: localhost
+  run_once: true
+  register: workflow_task_info
+'''
+
+from ansible.module_utils.ansible_tower import TowerModule, tower_auth_config
+
+try:
+    import tower_cli
+    from tower_cli.api import client
+    from tower_cli.conf import settings
+    from tower_cli.exceptions import ServerError, ConnectionError, BadRequest, TowerCLIError
+except ImportError:
+    pass
+
+
+def main():
+    argument_spec = dict(
+        workflow_template=dict(required=True),
+        extra_vars=dict(required=False),
+        wait=dict(required=False, default=True, type='bool'),
+        timeout=dict(required=False, default=None, type='int'),
+    )
+
+    module = TowerModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    workflow_template = module.params.get('workflow_template')
+    extra_vars = module.params.get('extra_vars')
+    wait = module.params.get('wait')
+    timeout = module.params.get('timeout')
+
+    # If we are going to use this result to return we can consider ourselfs changed
+    result = dict(
+        changed=False,
+        msg='initial message'
+    )
+
+    tower_auth = tower_auth_config(module)
+    with settings.runtime_values(**tower_auth):
+        # First we will test the connection. This will be a test for both check and run mode
+        # Note, we are not using the tower_check_mode method here because we want to do more than just a ping test
+        # If we are in check mode we also want to validate that we can find the workflow
+        try:
+            ping_result = client.get('/ping').json()
+            # Stuff the version into the results as an FYI
+            result['tower_version'] = ping_result['version']
+        except(ServerError, ConnectionError, BadRequest) as excinfo:
+            result['msg'] = "Failed to reach Tower: {0}".format(excinfo)
+            module.fail_json(**result)
+
+        # Now that we know we can connect, lets verify that we can resolve the workflow_template
+        try:
+            workflow = tower_cli.get_resource("workflow").get(**{'name': workflow_template})
+        except TowerCLIError as e:
+            result['msg'] = "Failed to find workflow: {0}".format(e)
+            module.fail_json(**result)
+
+        # Since we were able to find the workflow, if we are in check mode we can return now
+        if module.check_mode:
+            result['msg'] = "Check mode passed"
+            module.exit_json(**result)
```

Comment:
for a future enhancement, you could check `workflow['summary_fields']['user_capabilities']['start']`, and then give them a custom error message if it returns False, meaning they don't have permission to launch.